src/onyx/peer/task_lifecycle.clj

```diff
@@ -394,14 +394,21 @@
                                             window-id-state (get window-state window-id)
                                             [window-id-state' window-entries] (window-state-updates segment window-id-state window event grouping-fn)
                                             window-state' (assoc window-state window-id window-id-state')]
-                                        (list window-state' 
-                                              (conj log-entries window-entries))))
+                                        (list window-state' (conj log-entries window-entries))))
                                     (list (:state @window-state) [unique-id])
                                     windows)]
                         (state-extensions/store-log-entry state-log event ack-fn full-log-entry)
                         (swap! window-state assoc :state new-window-state))
-                      (doseq [t triggers]
-                        (triggers/fire-trigger! event window-state t {:segment segment :context :new-segment})))
+                      (let [trigger-entries
+                            (reduce
+                             (fn [entries t]
+                               (concat
+                                entries
+                                [unique-id
```

Comment:
We shouldn't store the unique-id a second time, it'll get restored to the filter twice which may cause some issues with bucketing.